metrics/web/test_driver/app_test.dart

```diff
@@ -157,6 +168,129 @@ void main() {
         expect(find.byType(AddProjectGroupCard), findsOneWidget);
       },
     );
+
+    testWidgets(
+      "creates a project group",
+      (WidgetTester tester) async {
+        await _pumpApp(tester);
+        await _openProjectGroupPage(tester);
+
+        await tester.tap(find.byType(AddProjectGroupCard));
+        await tester.pumpAndSettle();
+
+        final projectGroupNameInputFinder = find.widgetWithText(
+          MetricsTextFormField,
+          ProjectGroupsStrings.nameYourGroup,
+        );
+        await tester.enterText(projectGroupNameInputFinder, projectGroupName);
+
+        await tester.tap(find.byType(MetricsCheckbox).first);
+        await tester.pumpAndSettle();
+
+        final createButtonFinder = find.widgetWithText(
+          MetricsPositiveButton,
+          ProjectGroupsStrings.createGroup,
+        );
+        await tester.tap(createButtonFinder);
+        await tester.pumpAndSettle();
+
+        final projectGroupCardFinder = find.widgetWithText(
+          ProjectGroupCard,
+          projectGroupName,
+        );
+
+        expect(projectGroupCardFinder, findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "updates a project group",
+      (WidgetTester tester) async {
+        await _pumpApp(tester);
+        await _openProjectGroupPage(tester);
+
+        final projectGroupCardFinder = find.widgetWithText(
+          ProjectGroupCard,
+          projectGroupName,
+        );
+
+        await tester.ensureVisible(projectGroupCardFinder);
+
+        await _hoverWidget(tester, projectGroupCardFinder);
+
+        final editButtonFinder = find.descendant(
+          of: projectGroupCardFinder,
+          matching: find.widgetWithText(IconLabelButton, CommonStrings.edit),
+        );
+        await tester.tap(editButtonFinder);
+        await tester.pumpAndSettle();
+
+        final projectGroupNameInputFinder = find.widgetWithText(
+          MetricsTextFormField,
+          projectGroupName,
+        );
+        await tester.tap(projectGroupNameInputFinder);
+        await tester.enterText(
+          projectGroupNameInputFinder,
+          updatedProjectGroupName,
+        );
+        await tester.pumpAndSettle();
+
+        final saveButtonFinder = find.widgetWithText(
+          MetricsPositiveButton,
+          ProjectGroupsStrings.saveChanges,
+        );
+        await tester.tap(saveButtonFinder);
+        await tester.pumpAndSettle();
+
+        final updatedProjectGroupCardFinder = find.widgetWithText(
+          ProjectGroupCard,
+          updatedProjectGroupName,
+        );
+
+        await Future.delayed(const Duration(seconds: 2));
+
+        tester.takeException();
```

Comment:
It seems like we do not need to call it here because we are not expecting any exceptions in this test. See [documentation](https://api.flutter.dev/flutter/flutter_test/TestWidgetsFlutterBinding/takeException.html) for details.